core/src/replay_stage.rs

```diff
@@ -753,8 +753,8 @@ impl ReplayStage {
             .filter(|(_, stats)| stats.is_recent && !stats.has_voted)
             .collect();
 
-        //highest weight, lowest slot first
-        candidates.sort_by_key(|b| (b.1.weight, 0i64 - b.1.slot as i64));
+        //lowest weight, lowest slot first
+        candidates.sort_by_key(|b| (b.1.weight, b.1.slot));
```

Comment:
@aeyakovenko, @carllin does this look right?

Reply:
@pgarg66 aren't we more likely to have slot 8 propagated to all the nodes?

Reply:
I am seeing this test fail due to above condition.
https://github.com/solana-labs/solana/blob/master/local-cluster/tests/local_cluster.rs#L731

There is one node in the cluster, and it ends up skipping its own slots.

Reply:
Hmm, ok, I think you are right.